cylc/rose/utilities.py

```diff
@@ -33,7 +33,7 @@
 from metomi.rose.config_processor import ConfigProcessError
 from metomi.rose.env import env_var_process, UnboundEnvironmentVariableError
 
-if TYPE_CHECKING:
+if TYPE_CHECKING:  # pragma: no cover
```

Comment:
There's a better solution to this, we can configure patterns for coverage to exclude.

Here's how we skip these lines automatically in Cylc:

https://github.com/cylc/cylc-flow/blob/master/.coveragerc#L24-L39

```
exclude_lines =
    pragma: no cover


    # Don't complain if tests don't hit defensive assertion code:
    raise NotImplementedError
    return NotImplemented


    # Ignore type checking code:
    if (typing\.)?TYPE_CHECKING:
    @overload( |$)


    # Don't complain about ellipsis (exception classes, typing overloads etc):
    \.\.\.


    # Ignore abstract methods
    @(abc\.)?abstractmethod
```